09-features.md

```diff
@@ -64,10 +64,14 @@ The origin node:
     unless indicated that it must set the odd feature bit instead.
   * MUST NOT set feature bits it does not support.
   * MUST NOT set feature bits in fields not specified by the table above.
+  * MUST NOT set both the optional and mandatory field.
```

Comment:
Sender MUST / receiver SHOULD kind of makes sense to me here - don't do it, but if somebody does it's _fine_.

Reply:
I think it is a different way to read the spec, I read as @carlaKC did. I think that a node MUST not do it, but if there is a bug and the send will do it, the received should know how to handle it

Reply:
After reflecting on this for a long time, keeping with the principle of being restrictive with what you send and permissive in what you accept, I agree with you and am rescinding this suggestion. However, I do think that if this is the case we should change the requirement of the receiver away from

`SHOULD be treated as mandatory`

and changed to

`MUST NOT be treated as optional`

since this is the real requirement. If we leave it as SHOULD, it leaves a vacuum of what to do in the case where you do not heed the recommendation. Failing the negotiation, while annoying, is acceptable. However, it is not acceptable for the receiver to think that it is optional. I will add this suggestion to the document in the appropriate place.

Reply:
> If we leave it as SHOULD, it leaves a vacuum of what to do in the case where you do not heed the recommendation. 

Mh, this vacuum, to me, is okay. Also as discussed during the spec meeting last week, we do not have the requirements of what happens if both fields are specified because, in some cases, you can do smart things (but currently, we do not know what)

In addition, if you do not follow the raccomandation of the should, someone will alert you with an error message because you are building the news after this in the wrong way (most of the time)

I treat this as a dummy change and specify very trivial requirements that all the implementations are already doing. I need just this in lnprototest because in lnprototest there is a clear line of what is wrong and what is right to do in the protocol. So without these couple of line in lnprototest I can not write the test because there is no rules inside the spec

I would like to spend time hard think about more hard problems that currently we have in the spec